crates/astria-sequencer/src/app/mod.rs

```diff
@@ -429,134 +398,146 @@ impl App {
+            }
+            ProposalFingerprintData::Unset => {}
+            // We have a cached proposal from prepare proposal, but it does not match
+            // the current proposal. We should clear the cache and execute proposal.
+            //
+            // This can happen in HA nodes, but if happening in single nodes likely a bug.
+            ProposalFingerprintData::InvalidCheckedPrepared(_) => {
+                self.metrics.increment_process_proposal_skipped_proposal();
+                trace!(
+                    "there was a previously prepared proposal cached, but did not match proposal, \
+                     will clear and execute block"
+                );
+                self.update_state_for_new_round(&storage);
+            }
+            // There was a previously executed full block cached, likely indicates
+            // a new round of voting has started. Previous proposal may have failed.
+            ProposalFingerprintData::ExecutedBlock(_, proposal_hash)
+            | ProposalFingerprintData::InvalidCheckedExecutedBlock(_, proposal_hash) => {
+                if proposal_hash.is_none() {
+                    trace!(
+                        "there was a previously executed block cached, but no proposal hash, will \
+                         clear and execute"
+                    );
+                } else {
+                    trace!(
+                        "our prepared proposal cache executed fully, but did not committed, will \
+                         clear and execute"
+                    );
+                }
+            }
+            // No cached proposal, nothing to do for logging. Common case for validator voting.
+            ProposalFingerprintData::Unset => {}
         }
 
-        self.update_state_for_new_round(&storage);
-
-        let mut txs = VecDeque::from(process_proposal.txs.clone());
-        let received_rollup_datas_root: [u8; 32] = txs
-            .pop_front()
-            .ok_or_eyre("no transaction commitment in proposal")?
-            .to_vec()
-            .try_into()
-            .map_err(|_| eyre!("transaction commitment must be 32 bytes"))?;
-
-        let received_rollup_ids_root: [u8; 32] = txs
-            .pop_front()
-            .ok_or_eyre("no chain IDs commitment in proposal")?
-            .to_vec()
-            .try_into()
-            .map_err(|_| eyre!("chain IDs commitment must be 32 bytes"))?;
-
-        let expected_txs_len = txs.len();
-
-        let block_data = BlockData {
-            misbehavior: process_proposal.misbehavior,
-            height: process_proposal.height,
-            time: process_proposal.time,
-            next_validators_hash: process_proposal.next_validators_hash,
-            proposer_address: process_proposal.proposer_address,
-        };
+        // If we can skip execution just fetch the cache, otherwise need to run the execution.
+        let tx_results = if skip_execution {
```

Comment:
Moved logic to deduplicate the flow of calling post execute. Makes the diff a bit ugly unfortunately.